### PR TITLE
feat(functions): add script to assign stations to coordinate-only logs

### PR DESCRIPTION
**Pull request by Irishsmurf**
## 📋 Description

Fuel logs that have coordinates but no `stationId` — including station locations that were pinned retroactively by hand — have nothing to group on, so `FuelMapPage` falls back to rounding their raw coordinates and renders them as loose **"unassigned"** clusters.

This adds a one-off, server-side backfill job that resolves each such log to a real station the same way live logging does, clearing the unassigned clusters. It is the batch equivalent of the existing in-app **Profile → Maintenance → Identify Stations** action (`src/utils/migrationService.ts`), but runs over every user at once with admin credentials.

## 🚀 Proposed Changes

- Added `functions/src/scripts/assignStationsToLogs.ts`: for each log with coordinates and no `stationId`, looks up the nearest `amenity=fuel` via the OSM Overpass API, gets-or-creates the matching `stations` document, writes its id back onto the log, and folds the log's price into the station's running average. Existing stations are never overwritten and already-assigned logs are skipped, so it is safe to re-run.
- Mirrored the `backfillOdometer` conventions: pure, unit-tested helpers (`selectUnassignedLogs`, `pickNearestStation`, `stationDocId`, `pricePerLitre`, `haversineDistanceKm`) separated from the Firestore/Overpass I/O, plus a `require.main` guard.
- CLI flags: `--dry-run`, `--user=<uid>`, `--project=<id>` (defaults to `fuelog-paddez`), `--radius=<m>` (default 150), `--delay=<ms>` (default 1100, to respect Overpass fair-use limits, with backoff on HTTP 429), and `--verbose`.
- Added the `assign-stations` npm script to `functions/package.json`.
- Added `functions/src/__tests__/assignStationsToLogs.test.ts` (15 tests covering selection, nearest-station pick incl. ways/relations and tag fallbacks, id sanitisation, and pricing edge cases).
- Documented usage in `scripts/README.md`.

## 🛡️ Checklist
- [x] I have verified that all unit tests pass locally (`npm run test`). — full `functions` suite: 83 passed.
- [ ] I have verified that the build succeeds locally (`npm run build`). — `tsc` fails on a pre-existing `TS5011` (missing `rootDir`) that also fails on `main`/clean tree; type-checks clean with `tsc --noEmit --rootDir src`.
- [x] I have added/updated unit tests for my changes.
- [x] I have followed the project's coding standards and naming conventions.
- [x] I have updated the documentation (`docs/`) if necessary. — updated `scripts/README.md`.
- [ ] New features are gated behind a Remote Config flag. — N/A; this is a one-off maintenance script, not a user-facing feature.

## 🔗 Related Issues/PRs
N/A

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_01WQxYdmgVAufn4dcGZTLeBb)_

### CODE REVIEW

File: functions/src/scripts/assignStationsToLogs.ts [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+      `\nNo logs found in project "${PROJECT_ID}".\n` +
+        '  - Confirm the credentials target this project (GOOGLE_APPLICATION_CREDENTIALS).\n' +
+        '  - Override the project with --project=<id> or GOOGLE_CLOUD_PROJECT=<id>.' +
+        (ONLY_USER ? `\n  - Confirm --user=${ONLY_USER} is the correct UID.` : ''),
+    );
+    return;
+  }
+
+  const toAssign = selectUnassignedLogs(logs);
+  console.log(`${toAssign.length} have coordinates but no station.`);
```
[the remark this answers is not in the document]
Comment (Irishsmurf) [reply functions/src/scripts/assignStationsToLogs.ts:334]: Leaving this one as `.get()` for now. It matches the sibling backfill job (`backfillOdometer.ts`), which also loads all logs into memory, and the dataset here is personal fuel logs (hundreds, not millions), so streaming would add complexity without a real benefit. Happy to switch to `.stream()` if these jobs ever need to scale to much larger collections.

---
_Generated by [Claude Code](https://claude.ai/code)_
